pkg/store/hintspb/hints.proto

```diff
@@ -18,6 +18,12 @@ option (gogoproto.goproto_unkeyed_all) = false;
 option (gogoproto.goproto_unrecognized_all) = false;
 option (gogoproto.goproto_sizecache_all) = false;
 
+message SeriesRequestHints {
+    /// include_blocks is the list of blocks that should be queried. Any other block loaded in the
+    /// store can be skipped if the list is non empty. If the list is empty, no filtering is applied.
+    repeated Block include_blocks = 1 [(gogoproto.nullable) = false];
```

Comment:
Crazy idea: To me it really looks like we might want.... BlockMatchers? no? (:

Essentially I am worried that you want to have more flexible filter logic at som point (regexp, != etc). Matcher for arbitrary labels (like we do use reliable) where for now we will have external labels PLUS __block_id (same as we use in relabelling) would do the work.

What do you think? (:

Reply:
I think it's a smart idea. Let me give it a try.